### PR TITLE
Docs 19122 msk guide updates

### LINKED ISSUE #17343: Improvement documentation for Kafka/MSK PrivateLink
### Documentation request

https://materialize.com/docs/ops/network-security/privatelink/

Here are a couple of suggestions to improve the docs:

1. The docs jump right into creating resources without giving a lot of context. Initially I was confused where the target group should be created. My AWS account or some other AWS account? I think it would be helpful to have a high level overview of the different resources and how they interact. Something similar [to this diagram](https://d2908q01vomqb2.cloudfront.net/b6692ea5df920cad691c20319a6fffd7a4a766b8/2020/05/26/GSMSKPrivateLink_rev1.png) from [this blog](https://aws.amazon.com/blogs/big-data/how-goldman-sachs-builds-cross-account-connectivity-to-their-amazon-msk-clusters-with-aws-privatelink/), but maybe not as detailed.
2. Step 1e requires to add ip addresses to the respective target groups. We could add some prescriptive guidance on how to obtain these IPs.
3. Step 2b is really vague. It would be more helpful to state that the security group of the Kafka cluster needs to allow traffic in from the IPs of the ENIs of the NLB (and how to find these ENIs).
4. Step 2 states that one should make sure that health checks are passing, but this will only happen once the target groups are associated with an TCP listener (after Step 4).
5. It would be nice to have some trouble shooting guidance. In particular for the following error message: `ERROR:  error registering kafka topic for sink: error fetching metadata when creating new topic sink_topic for sink: Meta data fetch error: BrokerTransportFailure (Local: Broker transport failure): BrokerTransportFailure (Local: Broker transport failure)`. This happens, e.g., if the `AVAILABILITY ZONE` is not (or incorrectly) specified in the connection and the NLB has cross zone load balancing disabled (which is the default).

### Related work

_No response_

### PR DESCRIPTION
**Pull request by bobbyiliev**
### Motivation

Refreshing the Kafka MSK and self-managed Kafka guides to provide end-to-end and correct guidance per the issue described [here](https://github.com/MaterializeInc/materialize/issues/19374).

Also, separating the Privatelink and SSH network security guides into separate reusable components in order to prevent content duplication as those instructions are re-used in a lot of separate guides.

Closes #17343 #19374 #19122

### CODE REVIEW

File: doc/user/layouts/shortcodes/network-security/ssh-host.md [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+);
+```
+
+## Configure the SSH bastion server
+
+The bastion host needs a **public key** to connect to the Materialize tunnel you
+created in the previous step.
+
+1. Materialize stores public keys for SSH tunnels. Use a `SELECT` statement to
+   return two public keys:
```
Comment (morsapaes) [inline doc/user/layouts/shortcodes/network-security/ssh-host.md:31]:
```suggestion
1. Materialize stores public keys for SSH tunnels in the system catalog. Query [`mz_ssh_tunnel_connections`](/sql/system-catalog/mz_catalog/#mz_ssh_tunnel_connections) to retrieve the public keys:
```

File: doc/user/layouts/shortcodes/network-security/ssh-host.md [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+## Prerequisites
+
+Before you begin, make sure you have access to a bastion host. You will need:
+
+* The bastion host IP address and port number
+* The bastion host username
+
+## Create an SSH tunnel connection
+
+In Materialize, use a `CREATE CONNECTION` statement to create an [SSH tunnel connection](/sql/create-connection/#ssh-tunnel) to the bastion server:
```
Comment (morsapaes) [inline doc/user/layouts/shortcodes/network-security/ssh-host.md:15]:
```suggestion
In Materialize, create an [SSH tunnel connection](/sql/create-connection/#ssh-tunnel) to the bastion host:
```

File: doc/user/layouts/shortcodes/network-security/ssh-host.md [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+
+```sql
+CREATE CONNECTION ssh_connection TO SSH TUNNEL (
+    HOST '<SSH_BASTION_HOST>',
+    USER '<SSH_BASTION_USER>',
+    PORT <SSH_BASTION_PORT>
+);
+```
+
+## Configure the SSH bastion server
```
Comment (morsapaes) [inline doc/user/layouts/shortcodes/network-security/ssh-host.md:25]:
```suggestion
## Configure the SSH bastion host
```

File: doc/user/layouts/shortcodes/network-security/ssh-host.md [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+    Test the connection from the bastion host to the Kafka cluster or PostgreSQL instance.
+
+    ```bash
+    telnet <KAFKA_BROKER_HOST> <KAFKA_BROKER_PORT>
+    telnet <POSTGRES_HOST> <POSTGRES_PORT>
+    ```
+
+    If the command hangs, double-check your security group and firewall settings. If the connection is successful, you can proceed to the next step.
+
+4. Verify the tunnel connection from your source to your SSH bastion host
```
Comment (morsapaes) [inline doc/user/layouts/shortcodes/network-security/ssh-host.md:73]:
```suggestion
4. Verify the SSH tunnel connection from your source to your bastion host:
```

File: doc/user/layouts/shortcodes/network-security/ssh-host.md [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+
+    ```bash
+    # Command for Linux
+    ssh -L 9092:kafka-broker:9092 <SSH_BASTION_USER>@<SSH_BASTION_HOST>
+    ```
+
+    Verify that you can connect to the Kafka broker or PostgreSQL instance via the tunnel:
```
Comment (morsapaes) [inline doc/user/layouts/shortcodes/network-security/ssh-host.md:80]:
```suggestion
    Verify that you can connect to the Kafka broker or PostgreSQL instance via the SSH tunnel:
```

File: doc/user/layouts/shortcodes/network-security/ssh-host.md [hunk 6]
```diff
@@ -0,0 +1,127 @@
...
+    PermitTunnel yes
+    ```
+
+    Save the changes and restart the SSH service:
+
+    ```bash
+    sudo systemctl restart sshd
+    ```
+
+5. Retrieve the static egress IPs from Materialize and configure the firewall rules (e.g. AWS Security Groups) for your SSH bastion to allow SSH traffic for those IP addresses only.
```
Comment (morsapaes) [inline doc/user/layouts/shortcodes/network-security/ssh-host.md:107]:
```suggestion
5. Retrieve the static egress IPs from Materialize and configure the firewall rules (e.g. AWS Security Groups) for your bastion host to allow SSH traffic for those IP addresses only.
```

File: doc/user/layouts/shortcodes/network-security/ssh-host.md [hunk 7]
```diff
@@ -0,0 +1,127 @@
+Materialize can connect to data sources like Kafka, Confluent, and PostgreSQL with a
```
Comment (morsapaes) [inline doc/user/layouts/shortcodes/network-security/ssh-host.md:1]: Nit: could you rename the shortcode to be `ssh-tunnel`?

File: doc/user/content/ingest-data/kafka-self-hosted.md [hunk 8]
```diff
@@ -0,0 +1,111 @@
+---
+title: "Self-hosted Kafka"
```
Comment (morsapaes) [inline doc/user/content/ingest-data/kafka-self-hosted.md:2]:
```suggestion
title: "Ingest data from Self-hosted Kafka"
```

File: doc/user/content/ingest-data/kafka-self-hosted.md [hunk 9]
```diff
@@ -0,0 +1,111 @@
...
+description: "How to connect a self-hosted Kafka cluster as a source to Materialize."
+aliases:
+  - /integrations/aws-kafka/
+  - /integrations/amazon-kafka/
+  - /connect-sources/amazon-kafka/
+menu:
+  main:
+    parent: "kafka"
+    name: "Self-hosted Kafka"
```
Comment (morsapaes) [inline doc/user/content/ingest-data/kafka-self-hosted.md:11]:
```suggestion
    name: "Self-hosted"
```

File: doc/user/content/ingest-data/network-security/privatelink.md [hunk 10]
```diff
@@ -26,207 +26,21 @@ Materialize provides Terraform modules for both [MSK cluster](https://github.com
...
+
 {{< /tabs >}}
 
 ## Related pages
 
 - [`CREATE SECRET`](/sql/create-secret)
 - [`CREATE CONNECTION`](/sql/create-connection)
 - [`CREATE SOURCE`: Kafka](/sql/create-source/kafka)
-- [AWS MSK integration guide](/integrations/aws-msk)
+- [AWS Kafka integration guide](/integrations/aws-kafka)
```
Comment (morsapaes) [inline doc/user/content/ingest-data/network-security/privatelink.md:46]:
```suggestion
- Integration guides: [Self-hosted PostgreSQL](/ingest-data/postgres-self-hosted), [Amazon RDS for PostgreSQL](/ingest-data/postgres-amazon-rds), [Self-hosted Kafka](/ingest-data/kafka-self-hosted), [Amazon MSK](/ingest-data/amazon-msk)
```

**Review by morsapaes — CHANGES_REQUESTED** [review]
Thank you, Bobby! Made a bunch of purely formatting-related suggestions, so that everything is consistent (also with the PostgreSQL integrations guides).

**Review by dehume — APPROVED** [review]

**Review by morsapaes — APPROVED** [review]

**Pull request merged**